lib/logstash/plugin_mixins/jdbc.rb

```diff
@@ -81,14 +84,22 @@ def jdbc_connect
       :password => @jdbc_password.nil? ? nil : @jdbc_password.value,
       :pool_timeout => @jdbc_pool_timeout
     }.merge(@sequel_opts)
-    begin
-      Sequel.connect(@jdbc_connection_string, opts=opts)
-    rescue Sequel::PoolTimeout => e
-      @logger.error("Failed to connect to database. #{@jdbc_pool_timeout} second timeout exceeded.")
-      raise e
-    rescue Sequel::Error => e
-      @logger.error("Unable to connect to database", :error_message => e.message)
-      raise e
+    retry_attempts = @maximum_connection_attempts
+    loop do
+      retry_attempts -= 1
+      begin
+        return Sequel.connect(@jdbc_connection_string, opts=opts)
+      rescue Sequel::PoolTimeout => e
+        if retry_attempts <= 0
+          @logger.error("Failed to connect to database. #{@jdbc_pool_timeout} second timeout exceeded.")
+          raise e
+        end
+      rescue Sequel::Error => e
+        if retry_attempts <= 0
```

Comment:
I think we still want to log the error and say retrying connection. And after the retry attempts are exhausted, we can raise exception